src/Compilers/CSharp/Portable/Symbols/TupleTypeSymbol.cs

```diff
@@ -22,23 +23,70 @@ internal sealed class TupleTypeSymbol : NamedTypeSymbol, ITupleTypeSymbol
         internal const int RestPosition = 8; // The Rest field is in 8th position
 
         /// <summary>
-        /// Create a new TupleTypeSymbol from its declaration in source.
-        /// Parameter elementsNames may be uninitialized, or has to be fully populated.
+        /// Construct a TupleTypeSymbol from it's underlying ValueTuple type and element names.
+        ///
+        /// Parameter elementsNames has to empty or fully populated.
+        /// </summary>
+        private TupleTypeSymbol(
+            NamedTypeSymbol underlyingType,
+            ImmutableArray<string> elementNames,
+            AssemblySymbol accessWithin)
+        {
+            _underlyingType = underlyingType;
+
+            // build the fields
+            var fieldsBuilder = ArrayBuilder<TupleFieldSymbol>.GetInstance();
```

Comment:
I went back and forth on this. But I think you're right, in the common case (short tuples) we at least know the arity of underlyingType, so that estimate will work when names are not passed.